### PR TITLE
docs(aqueue): add README.mbt.md with verified examples

### PR DESCRIPTION
**Pull request by bobzhang**
## Summary

- Adds `src/aqueue/README.mbt.md` documentation covering queue kinds, close semantics, non-blocking operations, fan-in/fan-out/fairness patterns, and cancellation safety.
- Fixes a `Queue::get` / `Queue::close` handoff race: if `put` delivered a value directly to a suspended reader and `close()` ran before that reader resumed, `get` could raise `QueueAlreadyClosed` instead of returning the accepted value.
- Updates the fan-out documentation to use `close()` as the worker-pool end-of-stream signal, and clarifies zero-size buffers, soft vs hard close, `try_get` / `try_put` fairness, and cancellation cleanup with `defer`.

## Root cause

`try_put` can hand a value directly to a blocked reader by storing it in the reader slot and waking the coroutine. After suspension, `get` checked `self.closed` before reading that slot, so a same-round `close()` could turn an already-delivered value into a close error. `get` now returns `reader.value` first when a value was delivered, and only checks `closed` when no value is available.

## Test plan

- [x] `moon test src/aqueue` — 53 / 53 passing
- [x] `moon check src/aqueue`
- [x] `moon info src/aqueue`
- [x] `moon run -c` probe against cached `moonbitlang/async@0.19.1` reproduced the old fan-out loss with `close()` (8 / 9 items), while the local package test verifies the fix

### CODE REVIEW

**Comment by bobzhang** [comment]
```
[moonbitlang/async] test http/request_test.mbt:28 ("request cancel") failed: src/http/request_test.mbt:36:3-36:44@moonbitlang/async FAILED: `false` is not true
Total tests: 242, passed: 241, failed: 1.
Error: Process completed with exit code 2.
```

the CI failure is a flaky test not relevant to this PR

**Comment by Guest0x0** [comment]
> The fan-out example uses a typed sentinel (Queue[Int?]) rather than close() to terminate workers. This is a robust pattern that side-steps a corner case where close() runs in the same scheduling round as a put that just delivered a value to a now-suspended reader (the close-error check fires before the reader's value is returned).

In general, the order of things happening within the same scheduler round should not be relied on: the actual behavior can be either "as if" A happen first or "as if" B happen first. But if the actual behavior is something in-between, for example a `put` call succeed, but the put value never get retrieved after `close(clear=false)`, it should be considered a bug. Can you elaborate on this corner case?

**Comment by Guest0x0** [comment]
The CI failure is independent of this PR. It's a pattern I have not seen before. The behavior of that test relies of a remote server (`https://www.moonbitlang.com`) enabling `gzip` encoding, it seems this is no longer the case, perhaps it is some sort of CDN issue or something.

I cannot reproduce this locally. The result seems random on CI as well. I'll modify the test to not rely on an external server.

File: src/aqueue/README.mbt.md [hunk 1]
```diff
@@ -0,0 +1,567 @@
...
+### Fan-out: one producer, worker pool
+
+One producer feeds items; N worker tasks each call `get` and process
+items in parallel. The queue load-balances automatically because
+`get` is first-come-first-served.
+
+A robust way to signal "no more work" to a worker pool is to push one
+**sentinel** value per worker (here, `None`). This avoids any race
+between `close()` and an in-flight value, and the queue stays a
+plain typed pipeline.
```
Comment (Guest0x0) [inline src/aqueue/README.mbt.md]: `.close()` is introduced exactly to replace the sentinel pattern. It `.close()` cannot fulfill this goal, it should be considered a bug

File: src/aqueue/README.mbt.md [hunk 2]
```diff
@@ -0,0 +1,567 @@
...
+}
+```
+
+## Cancellation Safety
+
+When a task is cancelled while suspended inside `get`, the queue
+takes care to **not lose a value** that was already handed to it.
+If a producer has just delivered a value to the reader's slot and
+the reader is then cancelled, the value is still returned rather
+than silently dropped.
```
Comment (Guest0x0) [inline src/aqueue/README.mbt.md]: There is a simple principle behind this: when `get` is cancelled (i.e. fail with cancellation error), the result should be equivalent to the `get` call never happened. As a consequence, a cancelled `get` call should never consume any value, because that is not equivalent to "`get` never happened"

File: src/aqueue/README.mbt.md [hunk 3]
```diff
@@ -0,0 +1,567 @@
...
+    // No leftover value in the queue.
+    debug_inspect(q.try_get(), content="None")
+  })
+  json_inspect(log, content=["got 42"])
+}
+```
+
+A `get` that has *not* yet received a value, however, will raise the
+cancellation error normally — wrap it in `try ... catch` if you need
+to do cleanup.
```
Comment (Guest0x0) [inline src/aqueue/README.mbt.md]: since cancellation is just a normal error, `defer` should be mentioned here as well. In general users should seldom need to handle cancellation error specially

File: src/aqueue/README.mbt.md [hunk 4]
```diff
@@ -0,0 +1,567 @@
...
+  }
+  inspect(err, content="got MyDone")
+}
+```
+
+## Non-blocking Operations
+
+`try_get` and `try_put` give you synchronous, non-suspending access to
+the queue. They are useful for opportunistic draining, polling, or
+inside non-async contexts.
```
Comment (Guest0x0) [inline src/aqueue/README.mbt.md:348]: the fairness contract between `try_get`/`try_put` and blocked `get`/`put` should be mentioned here

File: src/aqueue/README.mbt.md [hunk 5]
```diff
@@ -0,0 +1,567 @@
...
+  })
+  json_inspect(received, content=["alpha", "beta", "gamma"])
+}
+```
+
+> **Tip:** When the producer wants to close immediately after the
+> last item, prefer a *bounded* queue. The forced suspension on full
+> guarantees that the consumer has caught up before `close()` runs,
+> avoiding the corner case where a value is in flight to a reader at
+> the instant the queue is closed.
```
Comment (Guest0x0) [inline src/aqueue/README.mbt.md]: The corner case also sounds like a bug here. To ensure immediate closure, bounded is not enough, because there is still a fixed buffer there. Immediate closure can only be guaranteed by either (1) `close(clear=true)` or (2) zero buffer size queue

File: src/aqueue/README.mbt.md [hunk 6]
```diff
@@ -0,0 +1,567 @@
...
+    "get() => 0", "put(0)", "get() => 1", "put(1)", "get() => 2", "put(2)",
+  ])
+}
+```
+
+### DiscardOldest(n)
+
+When the buffer is full, the **oldest** item is silently dropped to
+make room for the new one. `put` never blocks. Useful for "latest
+state" channels where stale data is uninteresting.
```
Comment (Guest0x0) [inline src/aqueue/README.mbt.md]: should mention: zero buffer size is allowed

File: src/aqueue/README.mbt.md [hunk 7]
```diff
@@ -0,0 +1,567 @@
...
+    result.push(x)
+  }
+  json_inspect(result, content=[2, 3])
+}
+```
+
+### DiscardLatest(n)
+
+When the buffer is full, the **incoming** item is silently dropped.
+`put` never blocks. Useful for sampling or rate limiting.
```
Comment (Guest0x0) [inline src/aqueue/README.mbt.md]: same here

File: src/aqueue/aqueue.mbt [hunk 8]
```diff
@@ -237,6 +237,9 @@ pub async fn[X] Queue::get(self : Queue[X]) -> X {
       raise err
     }
   }
+  if reader.value is Some(value) {
+    return value
+  }
```
Comment (bobzhang) [inline src/aqueue/aqueue.mbt]: cc @Guest0x0
